shell/packages/sandstorm-db/db.js

```diff
@@ -560,6 +566,52 @@ isSignedUpOrDemo = function () {
   return false;
 }
 
+var calculateReferralBonus = function(accountId, plan) {
+  // This function returns an object of the form:
+  //
+  // - {grains: 0, storage: 0}
+  //
+  // which are extra resources this account gets as part of participating in the referral
+  // program. (Storage is measured in bytes, as usual for plans.)
+
+
+  // Authorization note: Only call this if accountId is the current user!
+  var isPaid = (plan && plan !== "free");
+
+  successfulReferralsCount = countReferrals(accountId);
+  if (isPaid) {
+    var maxPaidStorageBonus = 30 * 1e9;
```

Comment:
This logic is very specific to Oasis, but quota is likely to be used much more broadly than that. This is e.g. why we have the `Plans` collection rather than hard-coding them. Would it make sense to store the referral parameters in `Plans`? For each plan type, you could store the max bonus and the per-referral bonus as parameters. If they aren't set, perhaps the referral program should turn itself off?